crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -474,44 +498,54 @@ impl<TX: DbTxMut + DbTx> DatabaseProvider<TX> {
                 }
             };
 
-            reverts
-                .entry(block_number)
+            revert_storage
+                .entry(block_and_address)
                 .or_default()
-                .entry(address)
-                .or_default()
-                .1
-                .push(old_storage);
+                .push((old_storage.key.into(), old_storage.value));
+
+            bundle_builder = bundle_builder
+                .state_storage(address, state_storage.entry(address).or_default().clone());
```

Comment:
I think we could probably avoid this clone if we had access to a mutable reference of `state_storage` on the `BundleBuilder`

Reply:
I don't mind, wdyt @joshieDo ?

Reply:
Cloning is very wasteful here, it is cloning full storage but only one slot is changed.

Making `get_.._mut` for builder fields makes sense.

Reply:
I'll update this whne https://github.com/bluealloy/revm/pull/1524 is merged.

Reply:
@rakita How should we manage upgrading revm version to include the use the mut getters? I tried modifiying the dependency to
 ```shell
revm = { git = "https://github.com/bluealloy/revm", rev = "8f4c153", features = [
    "std",
    "secp256k1",
    "blst",
], default-features = false }
``` 
but this caused a bunch of collison between the types because of https://github.com/paradigmxyz/revm-inspectors has a different dependency on different version of revm. Should I update it there first?

Reply:
Created a PR https://github.com/paradigmxyz/revm-inspectors/pull/143

Reply:
I've upgraded the dependencies and switch to use the mut getters.